### PR TITLE
Add/docker image caching

### LINKED ISSUE #5987: Document how often custom docker images are retrieved
# Missing Information

How often are job custom docker images pulled from a registry? 

I would like to use a custom docker image to run several jobs in a workflow. However it is not clear how often this image will be pulled from docker registry: 

- Will my custom image get pulled from our private docker registry every once for every job in the workflow? or just once for the whole workflow? 
- Will the image be cached somewhere in circleci so that if the workflow runs 100 times a day, the image will be fetched only once from its docker registry (then stored for some period of time in circleci somewhere, so that every time the jobs are run, circleci just fetches the image locally as done presumably for cimg/* images)

# Already Looked

I searched the circleci docs

# Potential Locations

In https://circleci.com/docs/2.0/custom-images/

### PR DESCRIPTION
**Pull request by rosieyohannan**
fixes #5987 

Use @BytesGuy 's support article to create a guide to how Docker images are pulled from the registry. Thank you @BytesGuy ! 🎉

### CODE REVIEW

**Comment by adamdmharvey** [comment]
Thanks, @rosieyohannan ! But for full transparency, that seems to be a different Adam than I, but I can't see that person's surname. :) So can't take any credit.

Anyways, great clarification to the docs! Thanks for helping the community.

**Comment by rosieyohannan** [comment]
omg oops! @BytesGuy 🎉  Thanks so much for this contribution! (Sorry @adamdmharvey I got this wrong, thanks for letting me know! 🙈 )

**Review by devinmarieb — APPROVED** [review]
@rosieyohannan nice! I would just resolve the bot issues. One seems to be missing the {: #caching-docker-images } header and one is just off syntax (it doesn't have a space between the `s` and the `}` I'm not sure if the syntax matters here, but weird it flagged it I guess.

File: jekyll/_cci2/custom-images.md [hunk 1]
```diff
@@ -515,3 +515,6 @@ docker push username/ruby-node:0.1
 
 To use the custom image, reference ruby-node/bar:0.1 in your `.circleci/config.yml` image key and your primary container will run it.
 It is worth it to commit your Dockerfile using a gist and link to it from Docker Hub to avoid losing your configuration.
+
+## Caching Docker images
```
Comment (rosieyohannan) [inline jekyll/_cci2/custom-images.md:519]:
```suggestion
## Caching Docker images
{: #caching-docker-images }
```

File: jekyll/_cci2/executor-types.md [hunk 2]
```diff
@@ -188,6 +188,20 @@ Capability | `docker` | `machine`
 
 For more information on `machine`, see the next section below.
 
+### Caching Docker images
```
Comment (rosieyohannan) [inline jekyll/_cci2/executor-types.md:191]:
```suggestion
### Caching Docker images
{: #caching-docker-images }
```

File: jekyll/_cci2/executor-types.md [hunk 3]
```diff
@@ -188,6 +188,20 @@ Capability | `docker` | `machine`
...
+{: caching-docker-images}
```
Comment (rosieyohannan) [inline jekyll/_cci2/executor-types.md:192]:
```suggestion
{: caching-docker-images }
```

**Pull request merged**